rest-mvc/src/main/java/com/intuit/tank/rest/mvc/rest/services/agent/AgentServiceV2Impl.java

```diff
@@ -0,0 +1,296 @@
+                configFile = agentConfigFile;
+            }
+            settings = FileUtils.readFileToString(configFile, StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            LOGGER.error("Error reading agent settings file: " + e);
+            throw new GenericServiceResourceNotFoundException("agent", "settings file", e);
+        }
+        return settings;
+    }
+
+    @Override
+    public File getSupportFiles() {
+        String filename = "agent-support-files.zip";
+        File supportFiles = new File(new TankConfig().getTmpDir(), filename);
+        if (!supportFiles.exists()) {
+            final FileStorage fileStorage = FileStorageFactory.getFileStorage(new TankConfig().getJarDir(), false);
+            final File harnessJar = new File(servletContext.getRealPath("/tools/" + HARNESS_JAR));
+            LOGGER.info("harnessJar = " + harnessJar.getAbsolutePath());
+            final List<FileData> files = fileStorage.listFileData("");
+
+            synchronized ( supportFiles ) {
+                supportFiles.getParentFile().mkdirs();
+                // open the zip stream in a try resource block, no finally needed
+                try (ZipOutputStream zip = new ZipOutputStream(new FileOutputStream(supportFiles))) {
+
+                    if (harnessJar.exists()) {
+                        addFileToZip(HARNESS_JAR, new FileInputStream(harnessJar), zip);
+                    }
+                    for (FileData fileData : files) {
+                        if (harnessJar.exists() && fileData.getFileName().equals(HARNESS_JAR)) {
+                            LOGGER.info("Not adding harness because we found it in the war.");
+                        } else {
+                            addFileToZip(fileData.getFileName(), fileStorage.readFileData(fileData), zip);
+                        }
+                    }
+                    zip.flush();
+                } catch (IOException e) {
+                    LOGGER.error("Error reading support files: " + e);
+                    throw new GenericServiceResourceNotFoundException("agent", "support files", e);
+                }
+            }
+        }
+        return supportFiles;
+    }
+
+    private void addFileToZip(String name, InputStream in, ZipOutputStream zip) {
+        Subsegment subsegment = AWSXRay.beginSubsegment("Zip.File." + name);
+        try {
+            zip.putNextEntry(new ZipEntry(name));
+            IOUtils.copy(in, zip);
+            zip.closeEntry();
+        } catch (IOException e) {
+            LOGGER.error(e.toString());
+            subsegment.addException(e);
+        } finally {
+            AWSXRay.endSubsegment();
+            if (in != null) {
+                try { in.close(); } catch (IOException e) {}
+            }
+        }
+    }
+
+    @Nonnull
+    @Override
+    public AgentTestStartData agentReady(AgentData data) {
+        AgentTestStartData response;
+        LOGGER.info("Agent ready: " + data);
```

Comment:
Is the X-Ray annotations done in the original agentReady being handled elsewhere? 
Source: https://github.com/intuit/Tank/blob/0593cc431d5ec4453fe77c45ef27d85bd1883b77/rest/service/agent/src/main/java/com/intuit/tank/service/impl/v1/agent/AgentServiceV1.java#L96

Reply:
Nope, added them to this agentReady, thanks for the heads up